### PR TITLE
improve schema check output & add tests

### PR DESCRIPTION
**Pull request by nr-kkenney**
Two main areas of change:
* adding in tests
* improving error output displayed

The specific improvements to error output in this pr are:
1. all errors are displayed. previously, if a file had multiple errors, only the first was displayed.
2. enum values are displayed.
3. nested properties that are required & missing display the path where they should live

### Example output
```none
Error: packs/apache/config.yml
	 must have required property 'name'
	 '/level' must be equal to one of the allowed values: ["New Relic","Verified","Community"]

Error: packs/apache/dashboards/apache.json
	 must have required property 'name'

Error: packs/elasticsearch/alerts/Cluster Health.yml
	 must have required property 'name'
	 must have required property 'type'
	 '/nrql/query' must have required property 'ball'
```

Resolves: #68

### CODE REVIEW

File: utils/validate_packs.js [hunk 1]
```diff
@@ -20,6 +21,33 @@ const EXCLUDED_DIRECTORY_PATTERNS = [
...
 ];
 
+/**
+ * Converts errors generated from ajv into 'general errors' we want to display to the user.
+ * @param {ErrorObject[]} ajvErrors - Errors generated from ajv validation.
+ * @return {Object[]} Array of our own internal error objects.
+ */
+const convertErrors = (ajvErrors) => {
+  const errors = ajvErrors.map((e) => {
+    console.log(e);
```
Comment (nr-kkenney) [inline utils/validate_packs.js:31]: dont need this line

File: utils/validate_packs.js [hunk 2]
```diff
@@ -92,24 +119,36 @@ const getPackFilePaths = (basePath) => {
...
-  console.log('');
+  printErrors(filesWithErrors);
 
   if (filesWithErrors.length > 0) {
     process.exit(1);
   }
 }
 
-main();
+if (require.main === module) {
```
Comment (nr-kkenney) [inline utils/validate_packs.js:150]: add a comment explaining this

File: utils/validate_packs.js (hunk 1 above)
Comment (rudouglas) [reply utils/validate_packs.js:31]:
```suggestion
```

File: utils/__tests__/validate_packs.test.js [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+      mainConfigTestFile.contents[0].level = 'INVALID_TYPE';
+
+      const { errors } = validateFile(mainConfigTestFile);
+
+      expect(errors.length).toBe(1);
+    });
+  });
+});
+
+describe('test converErrors', () => {
```
Comment (rudouglas) [inline utils/__tests__/validate_packs.test.js:301]: just a typo
```suggestion
describe('test convertErrors', () => {
```

**Review by rudouglas — APPROVED** [review]
LGTM apart from the superficial changes

**Review by rudouglas — APPROVED** [review]
LGTM

**Pull request merged**